### ISSUE TITLE
Add CLI provisioner flag for enabling leader election

### ISSUE THREAD

**Issue by timflannagan** [operator-framework/rukpak#577]
Placeholder for now: It looks like the core set of provisioners don't have a way to enable leader election mechanisms that are present in controller runtime.

**Comment by timflannagan** [comment]
Saw this while doing some downstream testing. I think we'd want to at least add a CLI flag toggling leader election in the short term. Ideally, this would be the default behavior, and you'd toggle off leader election during dev workflows.

**Comment by timflannagan** [comment]
FYI @joelanford

**Comment by akihikokuroda** [comment]
The provisioner executable has the **leader-elect** flag and its default is **false**
```
flag.BoolVar(&enableLeaderElection, "leader-elect", false,
		"Enable leader election for controller manager. "+
			"Enabling this will ensure there is only one active controller manager.")
```

The deployment.yaml doesn't  have the flag set. So it's false by default.
This flag don't have any effect with `replicas: 1` in the deployment.yaml.

**Comment by timflannagan** [comment]
Yep, you're right (and I'm blind). I _think_ leader election would still be relevant here as configuring a single replica for the provisioner deployment can result in new replicas being rolled out over time, and those new replicas could specify a bad container image, or that container image would result in runtime failures, etc. and the leader election mechanisms being enabled would help smooth out that transistion between provisioner pods.

**Comment by timflannagan** [comment]
@joelanford Just going through my open list of issues to see what can be closed and noticed this wasn't ever picked up. Might be worth tackling if this is still an issue before rukpak hits production.
